delete.php

```diff
@@ -16,14 +16,25 @@
 
 require_once('../../config.php');
 
-$cmid = required_param('cmid', PARAM_INT);
 $id = required_param('id', PARAM_INT);
 
-list ($course, $cm) = get_course_and_cm_from_cmid($cmid, 'cpdlogbook');
+// If the entry doesn't exist.
+if (! $record = $DB->get_record('cpdlogbook_entries', ['id' => $id, 'userid' => $USER->id])) {
```

Comment:
These bits of code can be a tiny bit simpler with this flag at the end:

$this->instance = $DB->get_record('assign', array('id'=>$update->id), '*', MUST_EXIST);